testsuite/features/support/twopence_init.rb

```diff
@@ -288,3 +281,66 @@ def file_inject(node, local_file, remote_file)
                   'sle12sp4_terminal'         => $sle12sp4_terminal,
                   'sle15sp3_buildhost'        => $sle15sp3_buildhost,
                   'sle15sp3_terminal'         => $sle15sp3_terminal }
+
+# This is the inverse of `node_by_host`.
+# For each node we choose as key the canonical host, i.e. we drop aliases.
+# For instance, we use 'ssh_minion` and ignore the alias 'ssh_spack_migrated_minion`.
+$host_by_node = {}
+$node_by_host.each do |host, node|
+  next if node.nil?
+
+  [host, node].each do |it|
+    raise ">>> Either host '#{host}' of node '#{node}' is empty.  Please check" if it == ''
+  end
+
+  ignored_hosts = %w[sle_ssh_tunnel_client sle_migrated_minion sle_spack_migrated_minion
+                     ssh_spack_migrated_minion sle_ssh_tunnel_minion
+                     ceos_ssh_minion ubuntu_ssh_minion]
+  next if ignored_hosts.include? host
+
+  $host_by_node[node] = host
+end
```

Comment:
Should we wrap that inside a function, and call it from there?

Reply:
This code is just executed once (It is run right after the declaration of `$node_by_host`):
to me, there is no benefect in wrapping it inside a function.

Reply:
oki